### PR TITLE
Add autodocumentation tools and decorators

### PR DESCRIPTION
**Pull request by SantiagoKsquare**
- Added autodocumentation tools,Schema documentation using Decorators on Model.
- new way to define routes using decorators on ModelController.
- constructor on ModelController using Decorators. 
- new modules on package.json.template.
- new path to see generated documentation.
- updated templates Generator with new routes definition using decorators.

### CODE REVIEW

**Review by rodksquare — COMMENTED** [review]
@SantiagoKsquare What is the file generators/app/templates/app/libraries.zip ?

**Comment by SantiagoKsquare** [comment]
> @SantiagoKsquare What is the file generators/app/templates/app/libraries.zip ?
@rodksquare  It is a file that I shared with my coworkers, I will delete it.

File: generators/app/templates/app/controllers/v1/Auth.ts [hunk 1]
```diff
@@ -44,42 +56,343 @@ export interface JWTPayload {
   email: string;
   role: User["role"];
 }
+@Controller("auth")
+// @ApiDocs()
```
Comment (rodksquare) [inline generators/app/templates/app/controllers/v1/Auth.ts]: Why is this commented?

Comment (SantiagoKsquare) [reply generators/app/templates/app/controllers/v1/Auth.ts]: > Why is this commented?
@rodksquare 
The auth controller doesn't have a model, so its requests and responses are not fully defined, so its documentation is optional or all requests and responses must be manually defined.

Comment (rodksquare) [reply generators/app/templates/app/controllers/v1/Auth.ts]: can we delete that line?

Comment (SantiagoKsquare) [reply generators/app/templates/app/controllers/v1/Auth.ts]: yes we can , I will delete it.

**Review by rodksquare — APPROVED** [review]

**Pull request merged**